src/data-access/mappers/groupDataMapper.ts

```diff
@@ -0,0 +1,18 @@
+import { EntityDataMapper } from './entityDataMapper.js';
+import { BaseGroup, BaseGroupDB, Group, GroupDB, Permission } from '../../types/group.js';
+
+
+export class GroupDataMapper extends EntityDataMapper {
+    toDomain(groupEntity: Group): Group {
```

Comment:
what is the purpose of this transformation? We return the same object structure as we receive in parameters (`Group`).

The purpose of mapper is to change the object structure.

There are a few object types:

1. DTO - to communicate between the internal business services. In the end, your service might return to controller, objects in DTO format.
2. RAW, ENTITY or DB - to retrieve, update and insert data into database. FORBIDDEN to use on higher than repository level.
3. CLIENT or DOMAIN - to communicate with client /response or request/

i.e.  if we want to get data from client and create entity in database, the flow could like like the following:

a) create DTO object from data you receive from client
```
const groupDTO = new UserGroupDTOModel(request.body)
...
const group = GroupServiceinstance.create(groupDTO) // this method returns data in DTO

...
return this.success(res, GroupMapper.toClient(group))
```

from this example, we can see, that `GroupServiceinstance.create` receives parameter in DTO and returns also in DTO. The difference between IN and OUT is that returned value contains already all required relations and real identifiicator(as when you create DTO object using `UserGroupDTOModel` it should not fill the ID property, as this is the only way you can make a difference).

So, `GroupMapper.toClient` expects parameter to be like `UserGroupDTOModel` and return value in `IUserGroupModel` (or whatever you might name this interface).

The same is about the mapper to transform dto to db structure. So, it could look like this

```
class UserGroupMapper {
...
static toEntity(dto: UserGroupDTOModel): IUserGroupEntity {
 return { 
     name: dto.name
     ...
 }
}
...
}
```

and for example `IUserGroupEntity` and `IUserGroupModel` could look like

```
export interface IUserGroupEntity {
   _id: number
   group_name: string
   group_description?: string
}

export interface IUserGroupModel {
   id: number
   name: string
   description?: string
}
```

and finally, the dto object creator can be like this:

```
class UserGroupDTOModel extends SomeBaseImplementation {
   constructor(props: IUserGroupPayload) {
     super(props)
   }

  get name(): string {
     return this.props.name = name 
  }
   ...
}
```